replicated_kv_store/client.go

```diff
@@ -6,102 +6,115 @@ import (
 	"io/ioutil"
 	"log"
 	"net/http"
+	"time"
 
 	"github.com/krithikvaidya/distributed-dns/replicated_kv_store/protos"
 )
 
 //WriteCommand is called when the client sends the replica a write request.
 func (node *RaftNode) WriteCommand(operation []string) bool {
+	for {
```

Comment:
1. The for loop doesn't need to be for the entire method, you can just put the for loop for checking the node.commitIndex == node.lastApplied condition, breaking if true and iterating again after timeout if false. Will make the code a bit more understandable

2. The places where the mutex is acquired/released will need to be changed. Currently the mutex for client write requests is acquired in the handler methods in raft_server.go, and released below in this method. However, if node.lastApplied != node.commitIndex, the lock will have to be released while waiting for committed entries to get applied, so that ApplyToStateMachine() can acquire the lock and actually apply the committed entries. A readlock while checking node.state in the HTTP handler functions, which is immediately released after will be enough there, and the full lock can be obtained+released appropriately here. 

3. 100ms between iterations might be a bit much, around 20ms might be better

Similarly for the readcommand method.